src/sequence.js

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2018 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/**
+ * Turn any object into an iterator.
+ * Takes objects that implement the iterator protocol.
+ * Plain objects are treated as key-value stores and yield
+ * a sequence of their key value bytes, represented as size-2 arrays.
+ *
+ * Any value that is allowed as a parameter for this function shall be
+ * considered to be a `Sequence` for the purpose of this file.
+ * This term shall be distinguished from `Iterable` in that iterables
+ * must implement the iterator protocol `iterable[Symbol.iterator]()`.
+ *
+ * @param {Object|Iterable|Iterator} obj
+ * @returns {Iterator}
+ * @yields The data from the given elements
+ */
+const iter = obj => {
+  function *objIter() {
+    for (const key in obj) {
+      if (obj.hasOwnProperty(key)) {
+        yield [key, obj[key]];
+      }
+    }
+  }
+  return obj.constructor === Object ? objIter(obj) : obj[Symbol.iterator]();
+};
+
+/**
+ * Iterate over sequences: Apply the give function to
+ * every element in the sequence
+ *
+ * @param {Sequence} seq Any sequence for which iter() is defined
+ * @param {Function} fn Function taking a single parameter
+ */
+const each = (seq, fn) => {
+  // eslint-disable-next-line no-restricted-syntax
+  for (const val of iter(seq)) {
+    fn(val);
```

Comment:
maybe also pass an `idx` and `seq`. see [Array.forEach](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach) or [Map.forEach](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/forEach)

Reply:
TBH I would prefer to introduce an `enumerate()` function.
I have run into issues more than once with the extra parameters when passing a external functions taking optional second and third parameters as opposed to lambdas I just created.